src/main/java/org/sonar/plugins/github/GitHubPluginConfiguration.java

```diff
@@ -64,6 +64,9 @@ public String repository() {
       String url = settings.getString(CoreProperties.LINKS_SOURCES);
       repo = parseGitUrl(url);
     }
+    if (repo == null) {
+      throw MessageException.of("Unable to find GitHub repository name for this project. Please provide it using property '" + GitHubPlugin.GITHUB_REPO + "'.");
```

Comment:
Perhaps the message should include the case when it fails to parse. With this message, it seems to be an  "else" from the previous if.